### PR TITLE
Showing timestamps in messages

### LINKED ISSUE #320: Display message timestamps
As a **person who chats regularly** i want to be able to **see at which time a message was sent** to **know exactly when a conversation happened.**

### PR DESCRIPTION
**Pull request by jb3rndt**
Closes #320

### CODE REVIEW

File: packages/TelegramClient-Core.package/TCCMessage.class/class/timeStampToDateAndTime..st [hunk 1]
```diff
@@ -0,0 +1,4 @@
+json conversion
+timeStampToDateAndTime: anUnixTimestamp
```
Comment (phkeese) [inline packages/TelegramClient-Core.package/TCCMessage.class/class/timeStampToDateAndTime..st:2]: typo, `aUnixTimestamp`

File: packages/TelegramClient-UI.package/TCUMessage.class/instance/messageDateText.st [hunk 2]
```diff
@@ -0,0 +1,5 @@
+text conversion
+messageDateText
+
+	^ (String streamContents: [
+		:aStream | self date asTime print24: true showSeconds: false on: aStream ]), ' Uhr'
```
Comment (phkeese) [inline packages/TelegramClient-UI.package/TCUMessage.class/instance/messageDateText.st:5]: Why leave this in German when the rest of the app is in english?

**Review by phkeese — CHANGES_REQUESTED** [review]
Some requests, mostly little code quality and typo things

File: packages/TelegramClient-UI.package/TCUMessage.class/instance/messageDateText.st (hunk 2 above)
Comment (jb3rndt) [reply packages/TelegramClient-UI.package/TCUMessage.class/instance/messageDateText.st:5]: Do you suggest just removing the `' Uhr'` or do we also want a 12h format? I found a method that displays this in a 12h format which is pretty convenient but I am not sur if that is what the customer wants.. What do you think?

File: packages/TelegramClient-Tests.package/TCTChatTests.class/instance/setUp.st [hunk 3]
```diff
@@ -8,4 +8,5 @@ setUp
 			message: aNumber asString;
 			chatID: aNumber * 2;
 			senderID: aNumber * 3;
-			id: aNumber * 4)].
\ No newline at end of file
+			id: aNumber * 4;
+			date: 1000000000 + (aNumber * 234567890))].
```
[the remark this answers is not in the document]
Comment (jb3rndt) [reply packages/TelegramClient-Tests.package/TCTChatTests.class/instance/setUp.st]: Please see d430259 and say if that is better :)

[the remark this answers is not in the document]
Comment (Storyxx) [reply packages/TelegramClient-Tests.package/TCTChatTests.class/instance/setUp.st]: Doesn't date store a DateAndTime Object, why are we assigning seconds here?

[the remark this answers is not in the document]
Comment (jb3rndt) [reply packages/TelegramClient-Tests.package/TCTChatTests.class/instance/setUp.st]: Hm thats a good point... Solved in 61370a5

File: packages/TelegramClient-UI.package/TCUMessage.class/instance/messageDateText.st (hunk 2 above)
Comment (phkeese) [reply packages/TelegramClient-UI.package/TCUMessage.class/instance/messageDateText.st]: I suggest just removing the 'Uhr'. The official app does so as well in English

File: packages/TelegramClient-Tests.package/TCTChatTests.class/instance/setUp.st (hunk 3 above)
[the remark this answers is not in the document]
Comment (phkeese) [reply packages/TelegramClient-Tests.package/TCTChatTests.class/instance/setUp.st]: > Please see [d430259](https://github.com/hpi-swa-teaching/TelegramClient/commit/d4302597221b6ce11e145dfb612cdc20a4df9577) and say if that is better and still correct for your understanding :)

Yes, that makes the code much more readable

**Review by phkeese — APPROVED** [review]

**Comment by phkeese** [comment]
I think this can be merged

**Pull request merged**